src/server/api/data.js

```diff
@@ -0,0 +1,144 @@
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the GET /api/filesystem/root handler");
+    api.get(
+        "/filesystem/root",
+        auth.authnTokenMiddleware,
+        terrainHandler({ method: "GET", pathname: "/secured/filesystem/root" })
+    );
+
+    logger.info(
+        "adding the PATCH /api/filesystem/entry/:resourceId/tags handler"
+    );
+    api.patch(
+        "/filesystem/entry/:resourceId/tags",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "PATCH",
+            pathname: "/secured/filesystem/entry/:resourceId/tags",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info(
+        "adding the GET GET /api/filesystem/entry/:resourceId/tags handler"
+    );
+    api.get(
+        "/filesystem/entry/:resourceId/tags",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "GET",
+            pathname: "/secured/filesystem/entry/:resourceId/tags",
+        })
+    );
+
+    logger.info("adding the POST /api/filesystem/user-permissions handler");
+    api.post(
+        "/filesystem/user-permissions",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "POST",
+            pathname: "/secured/filesystem/user-permissions",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the POST /api/filesystem/delete handler");
+    api.post(
+        "/filesystem/delete",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "POST",
+            pathname: "/secured/filesystem/delete",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the POST /api/filesystem/directory/create handler");
+    api.post(
+        "/filesystem/directory/create",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "POST",
+            pathname: "/secured/filesystem/directory/create",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the GET /filetypes/type-list handler");
+    api.get(
+        "/filetypes/type-list",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "GET",
+            pathname: "/secured/filetypes/type-list",
+        })
+    );
+
+    logger.info("adding the POST /filetypes/type handler");
+    api.post(
+        "/filetypes/type",
+        auth.authnTokenMiddleware,
+        terrainHandler({
+            method: "POST",
+            pathname: "/secured/filetypes/type",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+    );
+
+    logger.info("adding the POST /share handler");
```

Comment:
Is this endpoint exclusive to data sharing?  I think we use this for apps and analyses sharing as well, which might make this hard to find in the `data` file

Reply:
Thats true. I will move this a separate file.